substrate/client/transaction-pool/tests/integration.rs

```diff
@@ -58,28 +51,40 @@ async fn send_future_and_then_ready_from_many_accounts() {
 	// Shared params.
 	let send_threshold = 20_000;
 	let ws = "ws://127.0.0.1:9933";
-	let recipe_future = TransactionRecipe::transfer();
-	let recipe_ready = recipe_future.clone();
 	let block_monitor = false;
-	let unwatched = false;
-
-	// Scenarios and sinks.
-	let scenario_future =
-		SendingScenario::FromManyAccounts { start_id: 0, last_id: 99, from: Some(100), count: 100 };
-	let scenario_ready =
-		SendingScenario::FromManyAccounts { start_id: 0, last_id: 99, from: Some(0), count: 100 };
+	let watched_txs = true;
 
-	let future_scenario_planner =
-		ScenarioPlanner::new(ws, scenario_future, recipe_future, block_monitor).await;
-	let ready_scenario_planner =
-		ScenarioPlanner::new(ws, scenario_ready, recipe_ready, block_monitor).await;
+	let future_scenario_executor = ScenarioBuilder::new()
+		.with_rpc_uri(ws.to_string())
+		.with_chain_type(ChainType::Sub)
+		.with_block_monitoring(block_monitor)
```

Comment:
I suspect we will duplicate this across many tests, so this could be common function:
```
fn zn_scenario_builder(ws) -> ScenarioBuilder {
	let send_threshold = 20_000;
	let block_monitor = false;
	let watched_txs = true;

ScenarioBuilder::new()
		.with_rpc_uri(ws.to_string())
		.with_chain_type(ChainType::Sub)
		.with_block_monitoring(block_monitor)
		.with_watched_txs(watched_txs)
		.with_send_threshold(send_threshold)
}

Reply:
Good idea, implemented here: [9a665b6](https://github.com/paritytech/polkadot-sdk/pull/7257/commits/9a665b669efab2e2fb89c836be62ade310eb695f)